src/State/helpers.js

```diff
@@ -25,3 +27,92 @@ export const transformRequests = (requests, role) => {
     }
   });
 };
+
+/* constructs a request config, then passing that to aws4 to sign
+ * host: the base url of the API
+ * method: the HTTP method of the request
+ * path: the specific path to be appended to the host (e.g. /users/requests)
+ * params: any query paramters the request needs. Pass in null if there is no query string parameters
+ * data: the body of the request. Pass in null if there is no body
+ */
+export const constructRequest = (host, method, path, params, data) => {
+  const {
+    AccessKeyId,
+    EmpEmail,
+    SamlHash,
+    SecretKey,
+    SessionToken,
+    UserKey
+  } = store.getState().currentUser;
+
+  let userSession = {
+    AccessKeyId,
+    EmpEmail,
+    SamlHash,
+    SecretKey,
+    SessionToken,
+    UserKey
+  };
+
+  // construct request
+  let request = {
+    host,
+    method,
+    url: `${host}${path}`,
+    path,
+    data: { ...userSession }, // data paramter needed for axios
+    body: JSON.stringify({ ...userSession }) // body parameter needed for aws
+  };
```

Comment:
Super sorry for getting nitpicky with variable declarations, but the base objects aren't reassigned to, so we can do `const` - especially with `userSession`

Actually - I also totally get why you destructure then restructure into `userSession`, but it's spread again on line 63 - maybe there's a cleaner way? Could just assign `userSession` the value from `store.getState().currentUser`, and then lines 74-79 need a `.nameOfProperty` appended on, but it's a little clearer to understand what happens up top at least :)

Reply:
so the `store.getState().currentUser` object also brings in the `role` object, which I don't want to pass into the body either. I had other approaches to doing this but this was the cleanest way I was able to approach it.

Reply:
Got it, cool 👍